processing/src/main/java/org/apache/druid/storage/StorageConnectorProvider.java

```diff
@@ -20,9 +20,11 @@
 package org.apache.druid.storage;
 
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
-import com.google.inject.Provider;
+
+import java.io.File;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
-public interface StorageConnectorProvider extends Provider<StorageConnector>
+public interface StorageConnectorProvider
 {
+  StorageConnector createStorageConnector(File tempDir);
```

Comment:
Could you please add java docs for this method ?
The current implementation just ignores this tempDir if the property is already set which might confuse the caller. 

Is it possible to not change this interface and have the implementation take  the "tempFIle" as the constructor param. Based on that, we can set the correct tempDir values.